groovy-mode.el

```diff
@@ -334,7 +296,178 @@ The function name is the second group in the regexp.")
   (defconst groovy-dollar-slashy-open-regex
     (rx "$/"))
   (defconst groovy-dollar-slashy-close-regex
-    (rx "/$")))
+    (rx "/$"))
+  (defconst groovy-declaration-keyword-regex
+    (rx
+     (* space)
```

Comment:
I suggest we use `symbol-start` rather than `space` to separate keywords. It's more robust.